lib/galaxy/webapps/galaxy/api/library_datasets.py

```diff
@@ -402,6 +402,8 @@ def load(self, trans, payload=None, **kwd):
         if path is None:
             raise exceptions.RequestParameterMissingException('The required attribute path is missing.')
         folder = self.folder_manager.get(trans, folder_id)
+        if isinstance(path, int):
+            path = str(path)
```

Comment:
I don't know, I think this should be a string when passed in from the client, and I wonder if this doesn't hide something else going wrong there.